apps/main/src/llamalend/features/market-list/drawers/MarketListFilterDrawer.tsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react'
+import { LlamaMarket, LlamaMarketsResult } from '@/llamalend/entities/llama-markets'
+import { Button, Grid, Stack } from '@mui/material'
+import { t } from '@ui-kit/lib/i18n'
+import { FilterIcon } from '@ui-kit/shared/icons/FilterIcon'
+import { FilterProps } from '@ui-kit/shared/ui/DataTable/data-table.utils'
+import { DrawerHeader } from '@ui-kit/shared/ui/DrawerHeader'
+import { SwipeableDrawer } from '@ui-kit/shared/ui/SwipeableDrawer'
+import { SizesAndSpaces } from '@ui-kit/themes/design/1_sizes_spaces'
+import { HiddenMarketsResetFilters } from '../chips/HiddenMarketsResetFilters'
+import { LlamaListFilterChips } from '../chips/LlamaListFilterChips'
+import { MarketTypeFilterChips } from '../chips/MarketTypeFilterChips'
+import { LendingMarketsFilters } from '../LendingMarketsFilters'
+
+const { Spacing } = SizesAndSpaces
+
+type Props = {
+  userHasPositions: LlamaMarketsResult['userHasPositions'] | undefined
+  hasFavorites: boolean | undefined
+  data: LlamaMarket[]
+  minLiquidity?: number
+  hiddenMarketCount?: number
+  resetFilters: () => void
+  hasFilters: boolean
+} & FilterProps<string>
+
+export const MarketListFilterDrawer = ({
+  userHasPositions,
+  hasFavorites,
+  data,
+  minLiquidity,
+  hiddenMarketCount,
+  resetFilters,
+  hasFilters,
+  ...filterProps
+}: Props) => {
+  const [open, setOpen] = React.useState(false)
```

Comment:
how about useSwitch? It would remove quite a fell inline callbacks (which aren't great for performance)